frame/fast-unstake/src/lib.rs

```diff
@@ -221,28 +234,23 @@ pub mod pallet {
 			let ctrl = ensure_signed(origin)?;
 
 			ensure!(ErasToCheckPerBlock::<T>::get() != 0, <Error<T>>::CallNotAllowed);
-
-			let ledger =
-				pallet_staking::Ledger::<T>::get(&ctrl).ok_or(Error::<T>::NotController)?;
-			ensure!(!Queue::<T>::contains_key(&ledger.stash), Error::<T>::AlreadyQueued);
+			let stash = T::Staking::stash_by_ctrl(&ctrl).map_err(|_| Error::<T>::NotController)?;
```

Comment:
for the sake of posterity, if we were to ignore controllers in this pallet and only allow stash to do operations, things would have been simpler. 

I.e. 

```
The dispatch origin of this call must be signed by the controller account, similar to
		/// `staking::unbond`.
```

is the root of evil. 

Perhaps we should already move to that direction, and either do it.

Reply:
@rossbulat WDYT?

Reply:
@kianenigma Probably best to do this move in both staking and fast-unstake to make the code less confusing?
Seeing how fast-unstake is already released with this interface anyway. We could make an issue for this though. 

What is the proposal here, passing the stash account directly?